src/java/org/apache/cassandra/db/compression/CompressionDictionaryDetailsTabularData.java

```diff
@@ -0,0 +1,304 @@
+         */
+        public void validate()
+        {
+            if (keyspace == null)
+                throw new IllegalArgumentException("keyspace not specified");
+            if (table == null)
+                throw new IllegalArgumentException("table not specified");
+            if (dictId <= 0)
+                throw new IllegalArgumentException("Provided dictionary id is lower than 0, it is '" + dictId + "'.'");
+            if (dict == null || dict.length == 0)
+                throw new IllegalArgumentException("Provided dictionary byte array is null or empty.");
+            if (dict.length > FileUtils.ONE_MIB)
+                throw new IllegalArgumentException("Imported dictionary can not be larger than " +
+                                                   FileUtils.ONE_MIB + " bytes, but it is " +
+                                                   dict.length + " bytes.");
+            if (kind == null)
+                throw new IllegalArgumentException("Provided kind is null.");
+
+            CompressionDictionary.Kind dictionaryKind;
+
+            try
+            {
+                dictionaryKind = CompressionDictionary.Kind.valueOf(kind);
+            }
+            catch (IllegalArgumentException ex)
+            {
+                throw new IllegalArgumentException("There is no such dictionary kind like '" + kind + "'. Available kinds: " + Arrays.asList(CompressionDictionary.Kind.values()));
+            }
+
+            if (dictLength <= 0)
+                throw new IllegalArgumentException("Size has to be strictly positive number, it is '" + dictLength + "'.");
+            if (dict.length != dictLength)
+                throw new IllegalArgumentException("The length of the provided dictionary array (" + dict.length + ") is not equal to provided length value (" + dictLength + ").");
+
+            int checksumOfDictionaryToImport = CompressionDictionary.calculateChecksum((byte) dictionaryKind.ordinal(), dictId, dict);
+            if (checksumOfDictionaryToImport != dictChecksum)
+            {
+                throw new IllegalArgumentException(format("Computed checksum of dictionary to import (%s) is different from checksum specified on input (%s).",
+                                                          checksumOfDictionaryToImport,
+                                                          dictChecksum));
+            }
+        }
+
+        @Override
+        public CompressionDictionaryDataObject clone()
+        {
+            try
+            {
+                CompressionDictionaryDataObject clone = (CompressionDictionaryDataObject) super.clone();
+                if (this.dict != null)
+                {
+                    clone.dict = this.dict.clone();
+                }
+                return clone;
+            }
+            catch (CloneNotSupportedException e)
+            {
+                throw new AssertionError("Clone not supported", e);
+            }
+        }
+    }
```

Comment:
Why is this method and `Cloneable` necessary? Looks like it is only used in test code `org.apache.cassandra.db.compression.CompressionDictionaryDataObjectTest#assertInvalid`

Reply:
Yes, just for tests ... I think this is simple enough solution for what I want to achieve (have a copy of an object fast) but if you see it not right, I have reworked it to use builder.